src/main/java/qna/service/QnaService.java

```diff
@@ -39,28 +29,14 @@ public Question findQuestionById(Long id) {
     }
 
     @Transactional
-    public void deleteQuestion(User loginUser, Long questionId) throws CannotDeleteException {
+    public void deleteQuestion(User loginUser, Long questionId) {
         Question question = findQuestionById(questionId);
-        if (!question.isOwner(loginUser)) {
-            throw new CannotDeleteException("질문을 삭제할 권한이 없습니다.");
-        }
+        DeleteHistories deleteHistories = question.deleteAndCreateDeleteHistories(loginUser);
 
-        List<Answer> answers = answerRepository.findByQuestionIdAndDeletedFalse(questionId);
-        for (Answer answer : answers) {
-            if (!answer.isOwner(loginUser)) {
-                throw new CannotDeleteException("다른 사람이 쓴 답변이 있어 삭제할 수 없습니다.");
-            }
-        }
+        saveDeleteHistories(deleteHistories);
```

Comment:
```suggestion
        deleteHistoryService.saveAll(deleteHistories.getValues());
```

취향 차이일 수 있지만 다음과 같은 코드로 충분히 의미가 전달될 것 같아요 ㅎㅎ